napari/utils/_tests/test_key_bindings.py

```diff
@@ -218,13 +221,43 @@ def test_handle_single_keymap_provider():
     assert not hasattr(foo, 'C')
 
 
+def test_bind_user_key():
+    foo = Foo()
+    bar = Bar()
+    handler = KeymapHandler()
+    handler.keymap_providers = [bar, foo]
+
+    x = 0
+
+    @_bind_user_key('D')
+    def abc():
+        nonlocal x
+        x = 42
+
+    print(handler.keymap_chain)
+
+    assert handler.active_keymap == {
+        'A': types.MethodType(foo.class_keymap['A'], foo),
+        'B': types.MethodType(foo.keymap['B'], foo),
+        'D': abc,
+        'E': types.MethodType(bar.class_keymap['E'], bar),
+    }
+
+    handler.press_key('D')
+
+    _get_user_keymap().clear()
```

Comment:
why do the clear here?

Reply:
since it's a global keymap it affects the other tests

Reply:
lets try to avoid using global objects in tests at all.  you can mock it here with:

```python
from unittest.mock import patch

@patch('napari.utils.key_bindings.USER_KEYMAP', new_callable=dict)
def test_bind_user_key(keymap_mock):
    ...
```